src/services/GooglePlacesService.ts

```diff
@@ -131,6 +163,17 @@ class GooglePlacesService {
         return Promise.reject(err);
       });
   }
+
+  // helper function used to refresh place ID and call getRestaurantDetails again for NOT_FOUND error
+  refreshPlaceIDAndGetDetails(placeID: string): Promise<RestaurantDetails> {
+    return this.refreshPlaceID(placeID)
+      .then((newPlaceID: string) => {
+        return this.getRestaurantDetails(newPlaceID);
+      })
+      .catch((err: AxiosError) => {
+        return Promise.reject(err);
+      });
+  }
 }
```

Comment:
For this class, could you include visibility keywords at the beginning of each function? I think right now they're all implicitly public and I'm not sure if you wanted them to be!